### ISSUE TITLE
Is it possible to build with Visual Studio 2019?

### ISSUE THREAD

**Issue by Doko-Demo-Doa** [TypesettingTools/Aegisub#74]
It's been a long time since my last use of Aegisub (haven't touched Aegisub since 2016).

As I can see, this repo is a (not direct) fork of official Aegisub repo, but it contains several bug fixes that are not merged into official repo (yet? Hope so).

I came from web and mobile development with basic knowledge about C++ and Lua, and I would like to get my hands on it some times. Is it possible to build *this* repo with Visual Studio 2019 (with MSVC toolchain), or does it have to be GNU/MinGW, or does it have to be Visual Studio 2015?

Thanks in advance.

**Comment by CoffeeFlux** [comment]
It should be possible to build with VS 2019 once the Meson build system work is merged into master. It already builds on the meson-vs2019 branch, but there's some remaining work and the installer is not yet set up. Hopefully this will all be working and merged into master before the year is out.

**Comment by EleonoreMizo** [comment]
I built the master branch with VS 2019 without major issue. It mostly required to adjust the Windows SDK version to match what is actually installed, upgrade the platform toolset from 1.41 to 1.42 (probably not mandatory) and make the build system use the 2019 compiler instead of the 2017 one, so it can compile nicely a few recent C++ features here and there. However the `test` project does not compile, it’s just a few lines.

**Comment by Doko-Demo-Doa** [comment]
Can you please point out those lines? I mean, in the `test` project?

**Comment by EleonoreMizo** [comment]
```
2>------ Build started: Project: tests, Configuration: Release x64 ------
2>Aegisub\tests\tests\keyframe.cpp(44,2): error C2061: syntax error: identifier 'Error'
2>Aegisub\tests\tests\keyframe.cpp(44,2): error C2310: catch handlers must specify one type
2>Aegisub\tests\tests\keyframe.cpp(44,2): error C2311: 'overloaded function type': is caught by '...' on line 44
2>Aegisub\tests\tests\keyframe.cpp(45,2): error C2061: syntax error: identifier 'Error'
2>Aegisub\tests\tests\keyframe.cpp(45,2): error C2310: catch handlers must specify one type
2>Aegisub\tests\tests\keyframe.cpp(45,2): error C2311: 'overloaded function type': is caught by '...' on line 45
```

**Comment by CoffeeFlux** [comment]
Should build easily with Meson, instructions in README. I tested locally in 2019.
